### PR TITLE
fix: #2 keybind does not appear to work

### PR DESCRIPTION
**Pull request by LoneExile**

### CODE REVIEW

File: lua/neural.lua [hunk 1]
```diff
@@ -34,7 +34,11 @@ function NeuralPromptBorder()
 end
 
 function Neural.setup(options)
-    Config.setup(options)
+    if type(options) == "table" then
+        Config.setup(options)
+    else
+        return
```
Comment (Angelchev) [inline lua/neural.lua:40]: Might be worth throwing an error so a user knows their config is invalid

**Review by Angelchev — CHANGES_REQUESTED** [review]
Good idea, would be good to provide some feedback to the user

File: lua/neural.lua [hunk 2]
```diff
@@ -34,7 +34,12 @@ function NeuralPromptBorder()
 end
 
 function Neural.setup(options)
-    Config.setup(options)
+    if type(options) == "table" then
+        Config.setup(options)
+    else
+        print('Neural: config options must be a table')
```
Comment (Angelchev) [inline lua/neural.lua:40]: Consider writing to the error buffer with `vim.api.nvim_err_writeln('[Neural Error]:  ...')`

**Review by Angelchev — CHANGES_REQUESTED** [review]

File: lua/neural.lua [hunk 3]
```diff
@@ -228,6 +228,4 @@ function Neural.query(prompt, type, args)
     end
 end
 
-Neural.setup()
```
Comment (LoneExile) [inline lua/neural.lua:231]: The real issue at hand is that this was caused by the setup being performed twice.

**Review by LoneExile — COMMENTED** [review]
Is it still necessary to check type config user to table?

File: lua/neural.lua (hunk 3 above)
Comment (Angelchev) [reply lua/neural.lua:231]: Yeah that's the issue, good catch :+1:

**Review by Angelchev — APPROVED** [review]
Nice catch :+1:

**Pull request merged**